contracts/messaging/MessageBusReceiver.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.13;
+
+import "@openzeppelin/contracts-4.5.0/access/Ownable.sol";
+import "./interfaces/IAuthVerifier.sol";
+import "./interfaces/ISynMessagingReceiver.sol";
+import "forge-std/Test.sol";
+
+contract MessageBusReceiver is Ownable, Test {
+    address public authVerifier;
+
+    enum TxStatus {
+        Null,
+        Success,
+        Fail,
+        Retry
+    }
+
+    // Store all successfully executed messages
+    mapping(bytes32 => TxStatus) public executedMessages;
+
+    // TODO: Rename to follow one standard convention -> Send -> Receive?
+    event Executed(
+        bytes32 msgId,
+        TxStatus status,
+        address indexed _dstAddress,
+        uint64 srcChainId,
+        uint64 srcNonce
+    );
+    event NeedRetry(bytes32 indexed msgId, uint64 srcChainId, uint64 srcNonce);
+    event CallReverted(string reason);
+
+    constructor(address _authVerifier) {
+        authVerifier = _authVerifier;
+    }
+
+    function computeMessageId(
+        uint256 _srcChainId,
+        bytes32 _srcAddress,
+        address _dstAddress,
+        uint256 _nonce,
+        bytes calldata _message
+    ) public view returns (bytes32) {
+        return
+            keccak256(
+                abi.encode(
+                    _srcChainId,
+                    _srcAddress,
+                    block.chainid,
+                    _dstAddress,
+                    _nonce,
+                    _message
+                )
+            );
+    }
+
+    function getExecutedMessage(bytes32 _messageId)
+        public
+        view
+        returns (TxStatus)
+    {
+        return executedMessages[_messageId];
+    }
+
+    /**
+     * @notice Relayer executes messages through an authenticated method to the destination receiver
+     based on the originating transaction on source chain
+     * @param _srcChainId Originating chain ID - typically a standard EVM chain ID, but may refer to a Synapse-specific chain ID on nonEVM chains
+     * @param _srcAddress Originating bytes32 address of the message sender on the srcChain
+     * @param _dstAddress Destination address that the arbitrary message will be passed to
+     * @param _gasLimit Gas limit to be passed alongside the message, depending on the fee paid on srcChain
+     * @param _message Arbitrary message payload to pass to the destination chain receiver
+     */
+    function executeMessage(
+        uint256 _srcChainId,
+        bytes32 _srcAddress,
+        address _dstAddress,
+        uint256 _gasLimit,
+        uint256 _nonce,
+        bytes calldata _message,
+        bytes32 _messageId
```

Comment:
No need to pass `messageId` here, as it's computed anyways. Also, `IMessageBus` interface doesn't have `messageId` param.